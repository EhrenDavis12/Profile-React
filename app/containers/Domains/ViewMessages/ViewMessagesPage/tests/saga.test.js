/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
// import viewMessagesPageSaga from '../saga';

import rootSaga, {
  fetchUserStart,
  fetchUserSaga,
  fetchUserMessagesStart,
  fetchUserMessagesSaga,
  selectedUserMessagesStart,
  selectedUserMessagesSaga,
} from '../saga';

describe('viewMessagesPageSaga Saga', () => {
  describe('fetchUser saga group', () => {
    it('should execute the fetchUserSaga', () => {
      const generator = fetchUserSaga();
      const putDescriptor = generator.next(fetchUserStart).value;
      expect(putDescriptor).toMatchSnapshot();
    });

    it('should execute the fetchUserStart', () => {
      const generator = fetchUserStart();
      const putDescriptor = generator.next().value;
      expect(putDescriptor).toMatchSnapshot();
    });
  });

  describe('fetchUserMessages saga group', () => {
    it('should execute the fetchUserMessagesSaga', () => {
      const generator = fetchUserMessagesSaga();
      const putDescriptor = generator.next(fetchUserMessagesStart).value;
      expect(putDescriptor).toMatchSnapshot();
    });

    it('should execute the fetchUserMessagesStart', () => {
      const generator = fetchUserMessagesStart();
      const putDescriptor = generator.next().value;
      expect(putDescriptor).toMatchSnapshot();
    });
  });

  describe('selectedUserMessages saga group', () => {
    it('should execute the selectedUserMessagesSaga', () => {
      const generator = selectedUserMessagesSaga();
      const putDescriptor = generator.next(selectedUserMessagesStart).value;
      expect(putDescriptor).toMatchSnapshot();
    });

    it('should execute the selectedUserMessagesStart', () => {
      const generator = selectedUserMessagesStart({
        selectedUserMessage: { uuid: 'test' },
      });
      const putDescriptor = generator.next(selectedUserMessagesStart).value;
      expect(putDescriptor).toMatchSnapshot();
    });
  });

  describe('rootSaga saga', () => {
    it('should execute the rootSaga', () => {
      const generator = rootSaga();
      const putDescriptor = generator.next().value;
      expect(putDescriptor).toMatchSnapshot();
    });
  });
});
