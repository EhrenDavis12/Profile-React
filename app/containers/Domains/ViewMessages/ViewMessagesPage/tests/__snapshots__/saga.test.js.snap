// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewMessagesPageSaga Saga fetchUser saga group should execute the fetchUserSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "app/ViewMessagesPage/REQUEST_USER",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`viewMessagesPageSaga Saga fetchUser saga group should execute the fetchUserStart 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "localhost:3001/api/v1/users/?uuid=2f67b460-6469-11e9-b132-2b223266dc25",
      [Function],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`viewMessagesPageSaga Saga fetchUserMessages saga group should execute the fetchUserMessagesSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "app/ViewMessagesPage/REQUEST_USER_MESSAGES",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`viewMessagesPageSaga Saga fetchUserMessages saga group should execute the fetchUserMessagesStart 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "localhost:3001/api/v1/messagesByUser/?userUuid=2f67b460-6469-11e9-b132-2b223266dc25",
      [Function],
      [Function],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`viewMessagesPageSaga Saga rootSaga saga should execute the rootSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "ALL": Array [
    GeneratorFunctionPrototype {
      "_invoke": [Function],
    },
    GeneratorFunctionPrototype {
      "_invoke": [Function],
    },
    GeneratorFunctionPrototype {
      "_invoke": [Function],
    },
  ],
}
`;

exports[`viewMessagesPageSaga Saga selectedUserMessages saga group should execute the selectedUserMessagesSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "app/ViewMessagesPage/SELECT_USER_MESSAGE",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`viewMessagesPageSaga Saga selectedUserMessages saga group should execute the selectedUserMessagesStart 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      [Function],
      [Function],
      Object {
        "uuid": "test",
      },
      [Function],
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
